platforms/quorum/configuration/roles/create/k8s_secrets/tasks/main.yaml

```diff
@@ -0,0 +1,59 @@
+##############################################################################################
+#  Copyright Accenture. All Rights Reserved.
+#
+#  SPDX-License-Identifier: Apache-2.0
+##############################################################################################
+
+#############################################################################################
+# This role creates secrets
+#############################################################################################
+# Checks if certificates are already created and stored in vault.
+- name: Check if ambassador and tls certs already created
+  shell: |
+    vault kv get -field=rootcapem {{ vault.secret_path | default('secretsv2') }}/{{ component_ns }}/crypto/{{ node_name }}/certs
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  register: certs
+  when: check == "ambassador_creds"
+  ignore_errors: yes
+
+# Gets the existing ambassador tls certs
+- name: Get ambassador and tls certs from Vault
```

Comment:
First check if the ambassador secret is already created, if not, then only fetch the certs from vault.